### PR TITLE
CORE-845 Users can create/edit a community

### PR DESCRIPTION
**Pull request by ash3rz**
NOTE: I left the name as "Communities" for now, but I still think perhaps "Community Collections" or just "Collections" would be good. Also, not sure if we ever reviewed the communities icon - I used a list like icon for now.

Desktop listing view (added create button)
image

Desktop sample community:
image

Mobile sample community:
image
image
image

Re-tag apps confirmation:
image

### CODE REVIEW

**Review by psarando — APPROVED** [review]
LGTM 🎉 

I just had a few non-blocker suggestions 👍

File: stories/axiosMock.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 import axiosInstance from "../src/common/getAxios";
 import MockAdapter from "axios-mock-adapter";
 
-const AXIOS_DELAY = 2000;
+const AXIOS_DELAY = 200;
```
[the remark this answers is not in the document]
Comment (ash3rz) [reply stories/axiosMock.js:10]: Oops! No 😊 my impatience for 2 seconds showing.

File: public/static/locales/en/communities.json [hunk 2]
```diff
@@ -1,9 +1,41 @@
...
+    "followCommunityText": "Following a community adds it to your My Communities list for easier access.",
+    "followCommunityTitle": "Follow {{ name }}",
+    "getCommunityFail": "Failed to fetch community details.  Please try again.",
+    "helpText": "Communities are a curated collection of public resources that you can create. You can also follow a community to see and easily access that community's custom list of resources in your My Communities list.",
+    "invalidCommunityName": "Community name cannot contain `{{invalidCharList}}`. Invalid characters: `{{invalidChars}}`",
     "myCommunities": "My Communities",
     "name": "Name",
     "noCommunities": "No communities",
+    "noExternalApps": "Adding Agave/HPC apps is not currently supported",
+    "noExternalAppsNote": "NOTE: $t(noExternalApps).",
```
[the remark this answers is not in the document]
Comment (ash3rz) [reply public/static/locales/en/communities.json:27]: Ah yeah, I found it here: https://www.i18next.com/translation-function/nesting

In hindsight, it probably would've been useful to put the feature name "Communities" in one string that gets referenced everywhere else so there's only one string that needs to get updated if we need to rename it 🤔

File: src/components/communities/form/Toolbar.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                    baseId={baseId}
+                    render={(onClose) => [
+                        unfollowEnabled && (
+                            <MenuItem
+                                key={buildID(baseId, ids.BUTTONS.UNFOLLOW_MI)}
+                                onClick={() => {
+                                    onClose();
+                                    setUnfollowCommunityDlgOpen(true);
+                                }}
+                            >
```
[the remark this answers is not in the document]
Comment (ash3rz) [reply src/components/communities/form/Toolbar.js:129]: Oops, yes, good catch!

**Review by sri2k1us — APPROVED** [review]
LGTM.
May be `import SupervisedUserCircleIcon from '@material-ui/icons/SupervisedUserCircle';` for Communities icon ? I Like the concept of people sticking together and that gives a sense of `community`.

**Comment by sri2k1us** [comment]
> LGTM.
> May be `import SupervisedUserCircleIcon from '@material-ui/icons/SupervisedUserCircle';` for Communities icon ? I Like the concept of people sticking together and that gives a sense of `community`.

Heh. NM. That might be confusing with `Teams` icon that we currently use. May be we can do custom icon.

**Comment by ash3rz** [comment]
Thanks for the reviews!  I'll merge this now and we can chat about icons and whether the name should change in the next DE-UX meeting.

**Pull request merged**